hack/observability/kube-state-metrics/metrics/machine.yaml

```diff
@@ -25,21 +25,43 @@
             failure_domain:
             - spec
             - failureDomain
-            internal_ip:
-            - status
-            - addresses
-            - "[type=InternalIP]"
-            - address
             provider_id:
             - spec
             - providerID
             version:
             - spec
             - version
-            containerRuntimeVersion:
+        type: Info
+    - name: status_nodeinfo
+      help: Node information about a machine.
+      each:
```

Comment:
Due to the discussion on slack:

I'd propose to keep the metrics at `capi_machine_info` metric and not introduce a `capi_machine_status_nodeinfo`. (excluding `internal_ip`).

The second metric seems to be more confusing for me.